polynote-server/src/test/scala/polynote/server/repository/NotebookRepositorySpec.scala

```diff
@@ -139,21 +142,29 @@ class NotebookRepositorySpec extends FreeSpec with Matchers with MockFactory wit
       }
       "createNotebook" - {
         "for relative paths in the root mount" in {
-          (root.createNotebook _).expects("foo", None).once().returning(ZIO.succeed("foo"))
-          tr.createNotebook("foo", None).runIO shouldEqual "foo"
+          (root.createNotebook _).expects("foo", None, None).once().returning(ZIO.succeed("foo"))
+          tr.createNotebook("foo", None, None).runIO shouldEqual "foo"
         }
         "for relative paths in the other mounts" in {
-          (mount1.createNotebook _).expects("foo", None).once().returning(ZIO.succeed("foo"))
-          tr.createNotebook("one/foo", None).runIO shouldEqual "one/foo"
+          (mount1.createNotebook _).expects("foo", None, None).once().returning(ZIO.succeed("foo"))
+          tr.createNotebook("one/foo", None, None).runIO shouldEqual "one/foo"
         }
 
         "for absolute paths in the root mount" in {
-          (root.createNotebook _).expects("foo", None).once().returning(ZIO.succeed("foo"))
-          tr.createNotebook("/foo", None).runIO shouldEqual "foo"
+          (root.createNotebook _).expects("foo", None, None).once().returning(ZIO.succeed("foo"))
+          tr.createNotebook("/foo", None, None).runIO shouldEqual "foo"
         }
         "for absolute paths in the other mounts" in {
-          (mount1.createNotebook _).expects("foo", None).once().returning(ZIO.succeed("foo"))
-          tr.createNotebook("/one/foo", None).runIO shouldEqual "one/foo"
+          (mount1.createNotebook _).expects("foo", None, None).once().returning(ZIO.succeed("foo"))
+          tr.createNotebook("/one/foo", None, None).runIO shouldEqual "one/foo"
+        }
+
+        "for paths creating from a template" in {
+          val nb = templateNB("foo_template")
+          (root.createNotebook _).expects("foo", None, Some("foo_template.ipynb")).once().returning(ZIO.succeed("foo"))
+          (root.loadNotebook _).expects("foo").once().returning(ZIO.succeed(nb))
+          tr.createNotebook("foo", None, Some("foo_template.ipynb")).runIO shouldEqual "foo"
+          tr.loadNotebook("foo").runIO shouldEqual nb
```

Comment:
Does this test exercise the changes you made? At a glance it looks like it's calling the mocked functions and asserting that the mocked output is correct, so I'm not clear whether your code is actually being executed 🤔 

Unless I'm forgetting something - it's been a while! If you stick a print statement in `createNotebook` and `loadNotebook`, does it get printed with this test?

Reply:
I think you're right - I'll take a look this afternoon and run it by Jeremy when we meet tomorrow. Scala tests are not my strong suit just yet...🙂